### PR TITLE
Enable Java SDK Distroless container image variants.

### PR DESCRIPTION
**Pull request by damondouglas**
This PR addresses #32818 adding Java SDK Distroless container image variants. **Future PR will add a GitHub workflows that runs `examplesJavaRunnerV2IntegrationTestDistroless`**

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://github.com/apache/beam/blob/master/CONTRIBUTING.md#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI or the [workflows README](https://github.com/apache/beam/blob/master/.github/workflows/README.md) to see a list of phrases to trigger workflows.

### CODE REVIEW

File: runners/google-cloud-dataflow-java/build.gradle [hunk 1]
```diff
@@ -273,6 +273,87 @@ def createRunnerV2ValidatesRunnerTest = { Map args ->
...
+      excludeTestsMatching 'org.apache.beam.sdk.io.gcp.bigtable.BigtableWriteIT.testE2EBigtableWriteWithEmptyMutationFailures'
+      excludeTestsMatching 'org.apache.beam.sdk.io.gcp.bigtable.BigtableWriteIT.testE2EBigtableWriteWithEmptyRowFailures'
+      excludeTestsMatching 'org.apache.beam.sdk.io.gcp.bigtable.BigtableWriteIT.testE2EBigtableWriteWithInvalidTimestampFailures'
+      excludeTestsMatching 'org.apache.beam.sdk.io.gcp.bigtable.BigtableWriteIT.testE2EBigtableWriteWithOversizedQualifierFailures'
+    }
+  }
+}
+
+tasks.register('buildAndPushDistrolessContainerImage', Task.class) {
+  def allowed = ["java17", "java21"]
```
[the remark this answers is not in the document]
Comment (damondouglas) [reply runners/google-cloud-dataflow-java/build.gradle:323]: @Abacn Yes. Java distroless only contains Java 17 and Java 21: https://github.com/GoogleContainerTools/distroless/tree/main/java#image-contents. If we wanted to support this we would have to build our own Java 8 and Java 11 using bazel to reuse https://github.com/GoogleContainerTools/distroless/blob/main/java/BUILD

File: runners/google-cloud-dataflow-java/build.gradle [hunk 2]
```diff
@@ -273,6 +273,87 @@ def createRunnerV2ValidatesRunnerTest = { Map args ->
   }
 }
 
+tasks.register('googleCloudPlatformRunnerV2DistrolessIntegrationTest', Test.class) {
```
[the remark this answers is not in the document]
Comment (damondouglas) [reply runners/google-cloud-dataflow-java/build.gradle]: I changed to use Dataflow Example test suite. Thank you @Abacn :-)

**Review by Abacn — APPROVED** [review]
Thank you!

File: runners/google-cloud-dataflow-java/build.gradle (hunk 2 above)
[the remark this answers is not in the document]
Comment (Abacn) [reply runners/google-cloud-dataflow-java/build.gradle]: sorry checked that DataflowV2 PostCommit runtime was 4h. 7h was the Dataflow V2 validation test suite, my memory was wrong. still long though. Thanks for the change!

**Pull request merged**